File: productengineering/persistence/src/main/java/br/com/todeschini/persistence/entities/enums/TipoPintura.java
package br.com.todeschini.persistence.entities.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum TipoPintura implements MappableEnum {

    ACETINADA(1, "Acetinada"),
    ALTO_BRILHO(2, "Alto brilho"),
    ACETINADA_VIDRO(3, "Acetinada vidro");

    private final Integer value;
    private final String label;

    public static TipoPintura fromValue(Integer value) {
        for (TipoPintura obj : TipoPintura.values()) {
            if (obj.value.equals(value)) {
                return obj;
            }
        }
        throw new IllegalArgumentException("Valor desconhecido: " + value);
    }

    @Override
    public int getValue() {
        return value;
    }
}

File: frontend/src/models/enums/tipoPintura.ts
export const DTipoPinturaEnum = {
    ACETINADA: { name: "ACETINADA", value: 1, label: "Acetinada" },
    ALTO_BRILHO: { name: "ALTO_BRILHO", value: 2, label: "Alto Brilho" },
    ACETINADA_VIDRO: { name: "ACETINADA_VIDRO", value: 3, label: "Acetinada Vidro" },
} as const;

export const getLabel = (name: string): string | undefined => {
    return Object.values(DTipoPinturaEnum).find(item => item.name === name)?.label;
};
